src/main/java/org/skife/jdbi/v2/ClasspathStatementLocator.java

```diff
@@ -22,17 +22,53 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.lang.reflect.Method;
 import java.util.Collections;
 import java.util.Map;
 import java.util.WeakHashMap;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.regex.Matcher;
 
 /**
  * looks for [name], then [name].sql on the classpath
  */
 public class ClasspathStatementLocator implements StatementLocator
 {
-    private final Map<String, String> found = Collections.synchronizedMap(new WeakHashMap<String, String>());
+    private static class CacheKey {
+        final String name;
+        final Class<?> sqlObjectType;
+        final Method sqlObjectMethod;
+
+        public CacheKey(String name, Class<?> sqlObjectType, Method sqlObjectMethod) {
+            this.name = name;
+            this.sqlObjectType = sqlObjectType;
+            this.sqlObjectMethod = sqlObjectMethod;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+
+            CacheKey cacheKey = (CacheKey) o;
+
+            if (name != null ? !name.equals(cacheKey.name) : cacheKey.name != null) return false;
+            if (sqlObjectType != null ? !sqlObjectType.equals(cacheKey.sqlObjectType) : cacheKey.sqlObjectType != null)
```

Comment:
These naked `if` blocks without braces are causing the build to fail. Need to add braces.

Alternatively, since we depend on Java 7, you could use `java.util.Objects.equals` to simplify this method:

    return Objects.equals(this.name, cacheKey.name)
        && Objects.equals(this.sqlObjectType, cacheKey.sqlObjectType)
        && Objects.equals(this.sqlObjectMethod, cacheKey.sqlObjectMethod);